File: halley-ecommerce/src/main/java/com/kodat/of/halleyecommerce/address/AddressService.java
package com.kodat.of.halleyecommerce.address;

import com.kodat.of.halleyecommerce.dto.address.AddressDto;
import org.springframework.security.core.Authentication;

import java.util.List;

public interface AddressService {
    List<AddressDto> getAllAddresses(Authentication connectedUser);

    AddressDto getAddressById(Long addressId, Authentication connectedUser);

    void createAddress(AddressDto addressDto, Authentication connectedUser);

    void updateAddressById(Long addressId, AddressDto addressDto, Authentication connectedUser);

    void deleteAddressById(Long addressId, Authentication connectedUser);
}

File: halley-next/src/services/addressService.js
import authHeader from "@/services/authHeader";
import apiClient from "@/services/axiosInstance";
import globalAxiosInstance from "@/api/globalAxiosInstance";




export const getAddressById = (addressId) => {
    const headers = authHeader();
    return apiClient.get(`/addresses/${addressId}`, {
        headers
    }).then((response) => response.data)
        .catch((error) => {
            console.log('Error getting address', error);
            throw error;
        });
};

export const getAllAddresses = () => {
    const headers = authHeader();
    return globalAxiosInstance.get('/addresses', {
        headers
    }).then((response) => response.data)
        .catch((error) => {
            console.log('Error getting addresses', error);
            throw error;
        });
};


export const createAddress = (addressDto) => {
    const headers = authHeader();
    return globalAxiosInstance.post('/addresses' , addressDto , {
        headers
    }).then(response => response.data)
        .catch((error) => {
            console.log('Error creating addresse', error);
            throw error;
        });
};


export const updateAddressById = (addressId, addressDto) => {
    const headers = authHeader();
    return apiClient.put(`/addresses/${addressId}`, addressDto, {
        headers
    }).then(response => response.data)
        .catch((error) => {
            console.log('Error updating address', error);
            throw error;
        });
};


export const deleteAddressById = (addressId) => {
    const headers = authHeader();
    return apiClient.delete(`/addresses/${addressId}`, {
        headers
    }).then(response => response.data)
    .catch((error) => {
        console.log('Error deleting address', error);
        throw error;
    });
};






